back/app/models/related_idea.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: related_ideas
+#
+#  id              :uuid             not null, primary key
+#  idea_id         :uuid             not null
+#  related_idea_id :uuid             not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#
+# Indexes
+#
+#  index_related_ideas_on_idea_id                      (idea_id)
+#  index_related_ideas_on_idea_id_and_related_idea_id  (idea_id,related_idea_id) UNIQUE
+#  index_related_ideas_on_related_idea_id              (related_idea_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (idea_id => ideas.id)
+#  fk_rails_...  (related_idea_id => ideas.id)
+#
+
+# RelatedIdea represents a relationship between two ideas.
+#
+# Currently, being "related" implicitly means that the +idea+ is a *copy* or
+# *clone* of the +related_idea+.
+#
+# In the future, we may add an explicit +relationship+ attribute (or similar) to this
+# model to support different types of relationships between ideas, such as:
+# "copy", "clone", "duplicate", "generated_from" (e.g., AI-generated), etc.
+class RelatedIdea < ApplicationRecord
```

Comment:
As said above, small suggestion to change this to `IdeaRelation`. More elegant in association names and more accurate.